yt/yt/core/misc/unittests/error_ut.cpp

```diff
@@ -421,6 +421,48 @@ TEST(TErrorTest, SerializationDepthLimit)
     }
 }
 
+TEST(TErrorTest, DoNotDuplicateOriginalErrorDepth)
+{
+    constexpr int Depth = 20;
+    auto error = TError(TErrorCode(Depth), "error");
+    for (int i = Depth; i >= 2; --i) {
+        error = TError(TErrorCode(i), "error") << std::move(error);
+    }
+
+    auto errorYson = ConvertToYsonString(error);
+    error = ConvertTo<TError>(errorYson);
+
+    // Due to reserialization error already contains "original_error_depth" attribute.
+    // It should not be duplicated after the next serialization.
+    error = TError(TErrorCode(1), "error") << std::move(error);
+
+    // Use intermediate conversion to test YSON parser depth limit simultaneously.
+    errorYson = ConvertToYsonString(error);
+    auto errorNode = ConvertTo<IMapNodePtr>(errorYson);
+
+    error = ConvertTo<TError>(errorYson);
+    for (int index = 0; index < ErrorSerializationDepthLimit - 1; ++index) {
```

Comment:
It's worth ASSERTing that Depth >= ErrorSerializationDepthLimit somewhere above.